src/Plugin/Field/FieldWidget/FileEntityBrowser.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\file_entity\Plugin\Field\FieldWidget\FileEditableWidget.
+ */
+
+namespace Drupal\file_entity\Plugin\Field\FieldWidget;
+
+use Drupal\Component\Utility\SortArray;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\entity_browser\Plugin\Field\FieldWidget\EntityReference;
+
+/**
+ * Entity browser file widget.
+ *
+ * @FieldWidget(
+ *   id = "file_entity_browser",
+ *   label = @Translation("Entity browser (file)"),
+ *   provider = "entity_browser",
+ *   multiple_values = TRUE,
+ *   field_types = {
+ *     "file",
+ *     "image"
+ *   }
+ * )
+ */
+class FileEntityBrowser extends EntityReference {
+
+  /**
+   * Due to the table structure, this widget has a different depth.
+   *
+   * @var int
+   */
+  protected static $deleteDepth = 3;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    $settings = parent::defaultSettings();
+    $settings['view_mode'] = 'thumbnail';
+    return $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = parent::settingsForm($form, $form_state);
+
+    $element['field_widget_display']['#access'] = FALSE;
+    $element['field_widget_display_settings']['#access'] = FALSE;
+
+
+    $element['view_mode'] = [
+      '#title' => t('File view mode'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('view_mode'),
+      '#options' => \Drupal::service('entity_display.repository')->getViewModeOptions('file'),
```

Comment:
I get it. OK, let's leave it like this. We can change this in the future if needed. EB should be more stable at that point.